tools/disassemble.py

```diff
@@ -0,0 +1,320 @@
+from uctypes import struct, addressof, LITTLE_ENDIAN, UINT16, UINT32
```

Comment:
It might be worth adding an MIT license and copyright to this file.

Reply:
Thanks for that comment. Would you suggest generally adding this to all files in the repo? Or is there something about the disassembler specifically that makes it better to add here?

We currently have a LICENSE file in the repository root stating this is licensed under MIT and copyright by those listed in the AUTHORS file.

I see Micropython itself has the MIT licence at the beginning of all its files. I guess that is better, in case a file is distributed on its own somehow. Perhaps I'll make a commit separate to the PR to add the licence and copyright to all files.

Reply:
> Would you suggest generally adding this to all files in the repo?

Yes.  That makes it very clear for anyone who copies the file what the license/copyright is.

You can add a short header using the `SPDX-License-Identifier` format.  Or a long one like in the `micropython` repo.



> Perhaps I'll make a commit separate to the PR to add the licence and copyright to all files.

That sounds good!

Reply:
I'ld rather dislike having the same (long) license text as a header in each file.

If there is a header in a file, I'ld rather expect it to roughly tell what's inside that file and after that there could be also a short notice identifying the license.

Reply:
Ok, so perhaps we'll go the `SPDX-License-Identifier` route? I'll look at how that works, and how to add the copyright notice.